### ISSUE TITLE
False positive for B035 dictionary comprehension uses static key

### ISSUE THREAD

**Issue by rouge8** [astral-sh/ruff#9842]
## Code

```python
from dataclasses import dataclass


@dataclass
class MyObject:
    name: str


def f(obj, vals):
    return {obj.name: val for val in vals}
```

## Command

```console
❯ ruff --isolated --select B t.py
t.py:12:13: B035 Dictionary comprehension uses static key: `obj.name`
Found 1 error.
```

## Ruff version

ruff 0.2.0

**Comment by charliermarsh** [comment]
Is that not a true positive? Every key is using the same value (`obj.name`), right?
